cmd/clusterawsadm/api/bootstrap/v1beta1/types.go

```diff
@@ -224,6 +224,9 @@ type AWSIAMConfigurationSpec struct {
 	// TODO: This field could be a pointer, but it seems it breaks setting default values?
 	// +optional
 	S3Buckets S3Buckets `json:"s3Buckets,omitempty"`
+
+	// AllowAssumeRole enables the sts:AssumeRole permission within the CAPA policies
+	AllowAssumeRole bool `json:"allowAssumeRole,omitempty"`
```

Comment:
As this is used by one of credential types, I would say we need this to be true by default. 

Do you think there is a scenario where a user of CAPA doesn't want to allow usage of the role assumption credential type?

Reply:
If we go with always having it enabled then we can probably just add the action here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go#L220

Reply:
Perhaps in scenarios where a user only has a single account they wouldnt need to use a role credential and could just rely on the ec2 instance profile where CAPA runs to do everything they may not want sts:AssumeRole.  🤷 I dont particularly feel opinionated one way or the other.  I mean personally if CAPA says we now require this role as part of our default set of permissions in order to support X functionality whether you use it or not... its fine by me.  You tell me what you want and I'll do it however the team decides.  There was a bit of discussion around this here by other team members: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4046